pkg/clusterresource/builder.go

```diff
@@ -158,9 +155,8 @@ func (o *Builder) Build() ([]runtime.Object, error) {
 
 	var allObjects []runtime.Object
 	allObjects = append(allObjects, o.generateClusterDeployment())
-	if !o.SkipMachinePoolGeneration {
-		allObjects = append(allObjects, o.generateMachinePool())
-	}
+	allObjects = append(allObjects, o.generateMachinePool())
```

Comment:
Can we please change this to check if the generated machine pool is nil so that we do not have to keep adding and removing `SkipMachinePoolGeneration`?
```suggestion
	if mp := o.generateMachinePool(); mp != nil {
		allObjects = append(allObjects, mp)
	}
```